runtime/src/bank.rs

```diff
@@ -4036,12 +4036,26 @@ impl Bank {
                 .feature_set
                 .is_active(&feature_set::enable_durable_nonce::id());
         let hash_queue = self.blockhash_queue.read().unwrap();
+        let last_blockhash = hash_queue.last_hash();
+        let next_durable_nonce =
+            DurableNonce::from_blockhash(&last_blockhash, separate_nonce_from_blockhash);
+        let uses_invalid_durable_nonce_hash = |blockhash: &Hash| -> bool {
+            let nonce_must_be_advanceable = self
+                .feature_set
+                .is_active(&feature_set::nonce_must_be_advanceable::ID);
+            enable_durable_nonce
+                && nonce_must_be_advanceable
+                && blockhash == next_durable_nonce.as_hash()
+        };
```

Comment:
Why this logic cannot go in `check_transaction_for_nonce`?

Reply:
I initially put it in `check_message_for_nonce` but didn't like that it would impact `get_fee_for_message`. I think that `check_transaction_for_nonce` is fine though. I will move it there.